src/components/Form/components/FormEmail/index.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { useForm } from 'react-hook-form'
+import { Errors, FormItem, Label, FormInput } from '../../styled'
+
+type Props = {
+	register: ReturnType<typeof useForm>['register']
+	errors: ReturnType<typeof useForm>['errors']
+	onChange: () => void
+}
+
+const emailRegex = /^[A-Z0-9._-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i
+
+const FormEmail: React.FC<Props> = ({ register, errors, onChange }) => (
+	<>
+		<FormItem>
+			<Label htmlFor="email">E-mail:</Label>
+			<FormInput
+				name="email"
+				type="email"
+				placeholder="email@domain.com"
+				ref={register({
+					required: 'Musíte zadat e-mail!',
+					pattern: {
+						value: emailRegex,
+						message: 'Neplatný formát e-mailové adresy!',
+						// TODO: Must be unique!
+					},
+					maxLength: {
+						value: 50,
```

Comment:
```suggestion
						value: 254,
```
https://7php.com/the-maximum-length-limit-of-an-email-address-is-254-not-320/

Reply:
Suggestion committed. :joy: I was too mean on users with really long e-mails. :joy: